File: src/main/java/victor/training/cleancode/kata/videostore/horror/Rental.java
package victor.training.cleancode.kata.videostore.horror;

import java.util.Objects;

public class Rental {

   private final Movie movie;

   private final Integer noDaysRented;

    public Rental(Movie movie, Integer noDaysRented) {
        this.movie = Objects.requireNonNull(movie);
        this.noDaysRented = Objects.requireNonNull(noDaysRented);
    }

    public Movie getMovie() {
        return movie;
    }

    public Integer getNoDaysRented() {
        return noDaysRented;
    }

    static int calculateRenterPoints(Movie each, int noDaysRented) {

        // add bonus for a two day new release rental
        if (each.getMovieCategory() != null &&
                (each.getMovieCategory() == MovieCategory.NEW_RELEASE)
                && noDaysRented > 1)
         return 2;

        return 1;
    }

     static double calculateAmountOfCurrentMovie(MovieCategory category,int noDaysRented){
        double thisAmount = 0;
        switch (category) {
            case REGULAR -> {
                thisAmount += 2;
                if (noDaysRented > 2)
                    thisAmount += (noDaysRented - 2) * 1.5;
            }
            case NEW_RELEASE -> thisAmount += noDaysRented * 3;
            case CHILDREN -> {
                thisAmount += 1.5;
                if (noDaysRented > 3)
                    thisAmount += (noDaysRented - 3) * 1.5;
            }
        }
        return thisAmount;
    }
}

File: cs/1_Original/Soat.CleanCode.VideoStore.Original/Rental.cs
namespace Soat.CleanCode.VideoStore.Original
{
    public class Rental
    {
        public int DaysRented { get; }
        public virtual Movie Movie { get; }

        public Rental(Movie movie, int daysRented)
        {
            Movie      = movie;
            DaysRented = daysRented;
        }

        public decimal GetAmmount()
        {
            var thisAmount = 0m;

            //determines the amount for each line
            switch (Movie.PriceCode)
            {
                case Movie.REGULAR:
                    thisAmount += 2;
                    if (DaysRented > 2)
                    {
                        thisAmount += (DaysRented - 2) * 1.5m;
                    }
                    break;
                case Movie.NEW_RELEASE:
                    thisAmount += DaysRented * 3;
                    break;
                case Movie.CHILDREN:
                    thisAmount += 1.5m;
                    if (DaysRented > 3)
                    {
                        thisAmount += (DaysRented - 3) * 1.5m;
                    }
                    break;
            }

            return thisAmount;
        }
    }
}
